src/components/LogOut/index.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from "react";
+import "./style.css";
+import { useHistory } from "react-router-dom";
+import { Modal, message } from "antd";
+import { useTranslation } from "react-i18next";
+import firebase from "../../firebaseConfig";
+
+const auth = firebase.auth();
+
+const LogOut = ({ setUserExist }) => {
+  const { t } = useTranslation();
+  const [modalVisible, setModalVisible] = useState(false);
+  const history = useHistory();
+
+  const showModal = () => {
+    setModalVisible(true);
+  };
+
+  const handleCancel = () => {
+    setModalVisible(false);
+  };
+
+  const handleOk = () => {
+    auth
+      .signOut()
+      .then(() => {
+        setModalVisible(false);
+        setUserExist(false);
+        message.success("Successfully logged out!");
+      })
+      .catch((error) => {
+        message.warning(error.message);
+      });
+    history.push("/");
+  };
+
+  return (
+    <>
+      <button className="logoutButton" type="button" onClick={showModal}>
+        {t("navbar.headers.header3")}
+      </button>
+      <Modal
+        title={t("navbar.headers.header3")}
+        visible={modalVisible}
```

Comment:
we can directly use the setstates here instead of nesting it in another function